training/model_management/src/azureml/model/mgmt/downloader/downloader.py

```diff
@@ -44,8 +43,30 @@ def __init__(self, model_id: str):
         """
         self._model_id = model_id
         self._model_uri = self.HF_ENDPOINT + f"/{model_id}"
-        self._model_info = None
         self._hf_api = HfApi(endpoint=self.HF_ENDPOINT)
+        self.is_valid_id = None
+        self._model_info = None
+
+    @property
+    def is_valid_id(self):
+        """Returns true if Hugging face model id is valid."""
+        return self._is_valid_id
+
+    @is_valid_id.setter
+    def is_valid_id(self, value):
```

Comment:
Why not include all of this in the `@property` above? It seems odd to create a setter for an `is...` property. You can set `self._is_valid_id` and then return it if set or do the lookup otherwise.